pkg/app/registry/static/static.go

```diff
@@ -148,23 +148,31 @@ func (b *reg) ListProviders(ctx context.Context) ([]*registrypb.ProviderInfo, er
 	return providers, nil
 }
 
-func (b *reg) ListSupportedMimeTypes(ctx context.Context) (map[string]*registrypb.AppProviderList, error) {
+func (b *reg) ListSupportedMimeTypes(ctx context.Context) ([]*registrypb.MimeTypeInfo, error) {
 	b.RLock()
 	defer b.RUnlock()
 
-	mimeTypes := make(map[string]*registrypb.AppProviderList)
+	res := []*registrypb.MimeTypeInfo{}
+	mtmap := make(map[string]*registrypb.MimeTypeInfo)
 	for _, p := range b.providers {
 		t := *p
 		t.MimeTypes = nil
 		for _, m := range p.MimeTypes {
-			if _, ok := mimeTypes[m]; ok {
-				mimeTypes[m].AppProviders = append(mimeTypes[m].AppProviders, &t)
+			if _, ok := mtmap[m]; ok {
+				mtmap[m].AppProviders = append(mtmap[m].AppProviders, &t)
 			} else {
-				mimeTypes[m] = &registrypb.AppProviderList{AppProviders: []*registrypb.ProviderInfo{&t}}
+				mtmap[m] = &registrypb.MimeTypeInfo{
+					AppProviders: []*registrypb.ProviderInfo{&t},
+					Ext:          "", // TODO fetch from config
+					Name:         "",
+					Description:  "",
+					Icon:         "",
+				}
+				res = append(res, mtmap[m])
```

Comment:
Are you sure this works? As far as I can say `res` would just have a list of empty `MimeTypeInfo` objects, each containing only a single app in the `AppProviders` field. The `append` should happen after we've iterated over all the apps